Lib/subprocess.py

```diff
@@ -412,6 +422,24 @@ def run(*popenargs, input=None, timeout=None, check=False, **kwargs):
             stdout, stderr = process.communicate()
             raise TimeoutExpired(process.args, timeout, output=stdout,
                                  stderr=stderr)
+        except KeyboardInterrupt:
+            if cleanup_timeout is None:
+                cleanup_timeout = 1.0
+                if timeout:
+                    remaining_timeout = timeout - (_time() - start_time)
+                    cleanup_timeout = max(0.0, remaining_timeout)
+
+            if cleanup_timeout == 0.0:
+                process.kill()
+                process.wait()
+                raise
+            try:
```

Comment:
I propose to close stdin, stdout and stderr here to notify the child process that we want it to stop.

Otherwise, the program can hang on writing into stdout if the stdout pipe is full.